backend/src/apiserver/server/api_converter.go

```diff
@@ -1747,8 +1748,15 @@ func toModelTasks(t interface{}) ([]*model.Task, error) {
 		runId := execSpec.ExecutionObjectMeta().Labels[util.LabelKeyWorkflowRunId]
 		namespace := execSpec.ExecutionNamespace()
 		createdAt := execSpec.GetCreationTimestamp().Unix()
+		// Get sorted node names to make the results repeatable
+		nodeNames := make([]string, 0, len(execSpec.Status.Nodes))
```

Comment:
Just wondering, for cases when `execSpec.Status == nil` or `execSpec.Status.Nodes == nil`, are they possible to happen? If so, I think it's a good idea to cover these edge cases.

Reply:
I believe we don't need this as the `Status` is node pointer. `nodeNames` slice will get appropriate capacity and length. Here is an example: https://go.dev/play/p/5xcfq5TT7b4